libs/langchain/langchain/utilities/duckduckgo_search.py

```diff
@@ -32,82 +34,79 @@ def validate_environment(cls, values: Dict) -> Dict:
+
+        with DDGS() as ddgs:
+            ddgs_gen = ddgs.news(
+                query,
+                region=self.region,
+                safesearch=self.safesearch,
+                timelimit=self.time,
+                max_results=self.max_results,
+            )
+            if ddgs_gen:
+                return [r for r in ddgs_gen]
 
-    def results(
-        self, query: str, num_results: int, backend: str = "api"
-    ) -> List[Dict[str, str]]:
+    def run(self, query: str) -> str:
+        """Run query through DuckDuckGo and return concatenated results."""
+        if self.source == "text":
+            results = self._ddgs_text(query)
+        elif self.source == "news":
+            results = self._ddgs_news(query)
+        else:
+            results = []
+
+        if not results:
+            return "No good DuckDuckGo Search Result was found"
+        return " ".join(r["body"] for r in results)
+
+    def results(self, query: str, max_results: int) -> List[Dict[str, str]]:
         """Run query through DuckDuckGo and return metadata.
 
         Args:
             query: The query to search for.
-            num_results: The number of results to return.
+            max_results: The number of results to return.
 
         Returns:
             A list of dictionaries with the following keys:
                 snippet - The description of the result.
                 title - The title of the result.
                 link - The link to the result.
         """
-        from duckduckgo_search import DDGS
-
-        with DDGS() as ddgs:
-            results = ddgs.text(
-                query,
-                region=self.region,
-                safesearch=self.safesearch,
-                timelimit=self.time,
-                backend=backend,
-            )
-            if results is None:
-                return [{"Result": "No good DuckDuckGo Search Result was found"}]
-
-            def to_metadata(result: Dict) -> Dict[str, str]:
-                if backend == "news":
-                    return {
-                        "date": result["date"],
-                        "title": result["title"],
-                        "snippet": result["body"],
-                        "source": result["source"],
-                        "link": result["url"],
-                    }
-                return {
-                    "snippet": result["body"],
-                    "title": result["title"],
-                    "link": result["href"],
-                }
-
-            formatted_results = []
-            for i, res in enumerate(results, 1):
-                if res is not None:
-                    formatted_results.append(to_metadata(res))
-                if len(formatted_results) == num_results:
-                    break
-        return formatted_results
+        if self.source == "text":
+            return [
+                {"snippet": r["body"], "title": r["title"], "link": r["href"]}
```

Comment:
why get rid of date and source? are those not guaranteed to be in results

Reply:
You're right. I added 'date' and 'source' to the results